src/ahp/prover.rs

```diff
@@ -334,9 +340,15 @@ impl<F: PrimeField> AHPForR1CS<F> {
 
         let msg = ProverMsg::EmptyMessage;
 
-        assert!(w_poly.degree() <= domain_h.size() - domain_x.size() + zk_bound - 1);
-        assert!(z_a_poly.degree() <= domain_h.size() + zk_bound - 1);
-        assert!(z_b_poly.degree() <= domain_h.size() + zk_bound - 1);
+        if w_poly.degree() > domain_h.size() - domain_x.size() + zk_bound - 1 {
```

Comment:
Let's restore all the asserts here: they mustn't fail, and if they do a software bug is occurring